### ISSUE TITLE
React native web will not compile with react-native-video

### ISSUE THREAD

**Issue by hellolgl** [TheWidlarzGroup/react-native-video#2719]
ERROR in ./node_modules/react-native-video/Video.js 55:7
Module parse failed: Unexpected token (55:7)
You may need an appropriate loader to handle this file type, currently no loaders are configured to process this file. See https://webpack.js.org/concepts#loaders
|   }
| 
>   seek = (time, tolerance = 100) => {
|     if (isNaN(time)) {throw new Error('Specified time is not a number');}
| 

package.json

"react-native-video": "^5.2.0",

**Comment by hueniverse** [comment]
We are removing support for DOM.

**Comment by Leena-Sharma-rsys** [comment]
Can you add support for react-native web instead of having React-native-dom as this library is not maintained now.

**Comment by haveamission** [comment]
@hueniverse React Native Web is **not** React Native DOM. React Native Web is much more strongly supported and used, including by large orgs such as Twitter.

**Comment by haveamission** [comment]
@Leena-Sharma-rsys Expo supports expo-av, which seems to have video support:

https://docs.expo.dev/versions/v46.0.0/sdk/av/

**Comment by realchrisolin** [comment]
How is "we are removing support for DOM" a valid resolution for this issue?

The 5.2.0 version of this plugin is effectively broken and throws this "Unexpected token (55:7)" error without any clear or intuitive resolution. Please provide a viable workaround or reopen this issue so a fix can be prioritized.

**Comment by freeboub** [comment]
@realchrisolin i agree, please provide a fix for this issue...

**Comment by joshlbaker** [comment]
Bump! Should I downgrade or what?

**Comment by evoactivity** [comment]
At some point in the future I am going to want react-native-web support. I'm willing to work on that myself, but it is 100% a spare time thing so if anyone else needs support and can work on it straight away go ahead. I'll keep an eye on this issue and update if I make any progress.

**Comment by framerate** [comment]
To anyone seeing this issue, what is the fix in 2024? 

Have some platform checks and load a different video player for the web?

**Comment by joshlbaker** [comment]
> To anyone seeing this issue, what is the fix in 2024?
> 
> Have some platform checks and load a different video player for the web?

Yep this is what my team had to do. Cleanest way is to create a `Video.web.tsx` file and add your web-only logic there.

```js
import {createElement} from 'react-native-web';

export default function Video({src, height, width}) {
  const attrs = {
    src,
    height,
    width,
    controls: true,
    autoPlay: false,
    ...
  };

  return createElement('video', attrs);
}
```

**Comment by freeboub** [comment]
@joshlbaker if you can open a pull request for that, it wouild be wonderfull !

**Comment by zoriya** [comment]
I wrote a video.web component that is mostly a 1to1 mapping of rnv props, if there is interest I can upstream it, but it does not contain all features of rnv (no ads or drm support for example).
I also need HLS and advanced subtitles support, so I use hls.js and some subtitle renderer in js. I think subtitles renderer don't need to be integrated here as long as a HTMLVideoElement ref is still available, but I think having hls built-in is good.

**Comment by KrzysztofMoch** [comment]
@zoriya you can contact me when you will have time and we can upstream this!

**Comment by zoriya** [comment]
I'll probably have the time at the end of the month, early next month. I think I'll stick with a vanilla video player and let users handle hls/special media themselves.

**Comment by NadiaMit** [comment]
Are there any updates?

**Comment by KrzysztofMoch** [comment]
There is PR for web support #3958 - I think I will have time to review and merge this soon

**Comment by arijitsarkar2k14** [comment]
The [PR](https://github.com/TheWidlarzGroup/react-native-video/pull/3958) to support `react-native-web` is still open. When are you guys planning to merge the PR?

**Comment by moskalakamil** [comment]
Hi Guys! Web support has been merged #3958

**Comment by zoriya** [comment]
Keep in mind that this is the first PR for web support and that it's missing a lot of features.

I plan to bring soon:
- hls support
- subtitle support (external & builtin, as many codecs as possible)

It's missing but I won't implement it:
- ads
- drm protection
- probably a lot of other features I'm forgetting
